js/src/product-feed/product-statistics/status-box/sync-status.js

```diff
@@ -66,7 +67,14 @@ function getSyncResult( {
  * @return {JSX.Element} The status for the Product Sync
  */
 function SyncStatus() {
-	const { data } = useAppSelectDispatch( 'getMCProductStatistics' );
+	const { data, hasFinishedResolution } = useAppSelectDispatch(
+		'getMCProductStatistics'
+	);
+
+	if ( ! hasFinishedResolution || ( hasFinishedResolution && ! data ) ) {
```

Comment:
It could be simplified to:

```suggestion
	if ( ! hasFinishedResolution || ! data ) {
```